R/tm_player_injury_history.R

```diff
@@ -48,19 +48,20 @@ tm_player_injury_history <- function(player_urls) {
       }
 
       season_injured <- tryCatch(pg %>% rvest::html_nodes("td:nth-child(1)") %>% rvest::html_text(),
-                              error = function(e) season_injured <- NA_character_)
+                              error = function(e) season_injured <- NA_character_) %>% .replace_empty_na()
       injury <- tryCatch(pg %>% rvest::html_nodes(".zentriert+ .hauptlink") %>% rvest::html_text(),
-                             error = function(e) injury <- NA_character_)
+                             error = function(e) injury <- NA_character_) %>% .replace_empty_na()
       injured_since <- tryCatch(pg %>% rvest::html_nodes(".hauptlink+ .zentriert") %>% rvest::html_text() %>% .tm_fix_dates(),
-                           error = function(e) injured_since <- NA_character_)
+                           error = function(e) injured_since <- NA_character_) %>% .replace_empty_na()
       injured_until <- tryCatch(pg %>% rvest::html_nodes(".zentriert+ td.zentriert") %>% rvest::html_text() %>% .tm_fix_dates(),
-                              error = function(e) injured_until <- NA_character_)
+                              error = function(e) injured_until <- NA_character_) %>% .replace_empty_na()
       duration <- tryCatch(pg %>% rvest::html_nodes(".zentriert+ td.rechts") %>% rvest::html_text(),
-                            error = function(e) duration <- NA_character_)
+                            error = function(e) duration <- NA_character_) %>% .replace_empty_na()
       games_missed <- tryCatch(pg %>% rvest::html_nodes(".wappen_verletzung") %>% rvest::html_text() %>% as.numeric() %>% suppressWarnings(),
-                              error = function(e) games_missed <- NA_integer_)
+                              error = function(e) games_missed <- NA_integer_) %>% .replace_empty_na()
       club <- tryCatch(pg %>% rvest::html_nodes("img") %>% rvest::html_attr("alt"),
-                        error = function(e) goals <- NA_character_)
+                        error = function(e) goals <- NA_character_) %>% .replace_empty_na()
```

Comment:
Using internal function `.replace_empty_na()` everywhere to ensure values return a NA and allow us to construct the final df output